### PR TITLE
Added base classes to implement offline mechanisms

### PR DESCRIPTION
**Pull request by juanignaciomolina**
# Added base classes to implement offline mechanisms

Added the following classes required to implement a basic offline mechanism and API requests collapsing.

* BaseCallCollapser
* ICallCollapser
* ICache
* IRepositoryCallback
* IUpdate
* Repository

### CODE REVIEW

File: networking/build.gradle [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+group = 'com.github.Wolox'
+
+android {
+    compileSdkVersion 25
+    buildToolsVersion '25.0.2'
+
+    defaultConfig {
+        minSdkVersion 16
+        targetSdkVersion 25
```
Comment (emalamela) [inline networking/build.gradle]: Libraries don't have `versionCode` and/or `versionName`?

File: networking/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="ar.com.wolox.wolmo.networking">
+
+    <application android:allowBackup="true"
+        android:label="@string/app_name"
+        android:supportsRtl="true"
+        android:theme="@style/AppTheme">
```
Comment (emalamela) [inline networking/src/main/AndroidManifest.xml]: Just a minor thing, we can self-enclose this xml by using `/>`.

File: networking/src/main/res/values/colors.xml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (emalamela) [inline networking/src/main/res/values/colors.xml]: Is this file needed?

File: networking/src/main/res/values/styles.xml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<resources>
```
Comment (emalamela) [inline networking/src/main/res/values/styles.xml]: Same as the other empty file :wink:.

**Review by SpicyCactuar — CHANGES_REQUESTED** [review]
Some minor changes. Looking gooooood!

File: networking/build.gradle (hunk 1 above)
Comment (juanignaciomolina) [inline networking/build.gradle]: Nope, that's for regular apps. 
Both `versionCode` and `versionName` are used for app's updates (not library updates, that is specified in the version of the libraries used in the gradle config file for each app).

File: networking/src/main/AndroidManifest.xml (hunk 2 above)
Comment (juanignaciomolina) [inline networking/src/main/AndroidManifest.xml]: Sure, np.

File: networking/src/main/res/values/colors.xml (hunk 3 above)
Comment (juanignaciomolina) [inline networking/src/main/res/values/colors.xml]: File removed!

File: networking/src/main/res/values/styles.xml (hunk 4 above)
Comment (juanignaciomolina) [inline networking/src/main/res/values/styles.xml]: File removed!

**Comment by juanignaciomolina** [comment]
@emalamela Added the requested changes in https://github.com/Wolox/wolmo-networking-android/pull/2

Since we have to include the changes in that PR as well, please review that PR (which is pointing to the branch of this PR) and once that it is ready to go, we merge that one into this one. After that, we can merge to master and we are done with the networking module. Do you agree? :)

**Review by SpicyCactuar — APPROVED** [review]
*Ripple effect!*

**Pull request merged**